youtube_dl/extractor/reddit.py

```diff
@@ -49,6 +51,58 @@ def _real_extract(self, url):
         }
 
 
+class RedditGIE(InfoExtractor):
+    _VALID_URL = r'(?:https?://(?:[^/]+\.)?reddit\.com/gallery/(?P<id>[^/?#&]+))'
+    _TEST = {
+        # from https://www.reddit.com/r/BostonTerrier/comments/j8foda/my_mookie/ -> https://www.reddit.com/gallery/j8foda/
+        'url': 'https://www.reddit.com/gallery/j8foda/',
+        'info_dict': {
+            'id': 'flmjm893p7s51',
+            'ext': 'jpg',
+            'title': 'My Mookie',
+            'entries': [
+                {'url': 'https://i.reddit.com/flmjm893p7s51.jpg'},
+                {'url': 'https://i.reddit.com/fhcwp0793p7s5.jpg'},
+            ]
+        },
+        'params': {
+            'format': 'bestvideo',
+        },
+    }
```

Comment:
This test is breaking and searching other extractors I wasn't able to figure out what I need to do.
I'd appreciate pointers to documentation/other extractors.

The error from `python test/test_download.py -v TestDownload.test_RedditG` is

```
test_RedditG (__main__.TestDownload): ... [RedditG] j8foda: Downloading JSON metadata
[download] Downloading playlist: j8foda
[RedditG] playlist j8foda: Collected 2 video ids (downloading 2 of them)
[download] Downloading video 1 of 2
[generic] flmjm893p7s51: Requesting header
[generic] flmjm893p7s51: Downloading webpage
ERROR: URL could be a direct video link, returning it as such.; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/home/five/Projects/youtube-dl/youtube_dl/YoutubeDL.py", line 806, in wrapper
    return func(self, *args, **kwargs)
  File "/home/five/Projects/youtube-dl/youtube_dl/YoutubeDL.py", line 827, in __extract_info
    ie_result = ie.extract(url)
  File "/home/five/Projects/youtube-dl/youtube_dl/extractor/common.py", line 532, in extract
    ie_result = self._real_extract(url)
  File "/home/five/Projects/youtube-dl/youtube_dl/extractor/generic.py", line 2479, in _real_extract
    self._downloader.report_warning(
  File "/home/five/Projects/youtube-dl/test/helper.py", line 271, in _report_warning
    real_warning(w)
  File "test/test_download.py", line 52, in report_warning
    raise ExtractorError(message)
youtube_dl.utils.ExtractorError: URL could be a direct video link, returning it as such.; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

ERROR
```

Which suggests I need to mark my entries as direct download links?